src/Microsoft.Health.Fhir.Tests.Common/IVersionSpecificFhirClient.cs

```diff
@@ -0,0 +1,141 @@
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Hl7.Fhir.ElementModel;
+using Microsoft.Health.Fhir.Core.Models;
+using Microsoft.Health.Fhir.Tests.Common.FixtureParameters;
+using Task = System.Threading.Tasks.Task;
+
+namespace Microsoft.Health.Fhir.Tests.Common
+{
+    public interface IVersionSpecificFhirClient
+    {
+        Format Format { get; }
+
+        (bool SecurityEnabled, string AuthorizeUrl, string TokenUrl) SecuritySettings { get; }
+
+        HttpClient HttpClient { get; }
+
+        Task RunAsUser(TestUser user, TestApplication clientApplication);
+
+        Task RunAsClientApplication(TestApplication clientApplication);
+
+        Task<FhirResponse<ResourceElement>> CreateAsync(ResourceElement resource);
+
+        Task<FhirResponse<ResourceElement>> CreateAsync(string uri, ResourceElement resource);
+
+        Task<FhirResponse<ResourceElement>> ReadAsync(string resourceType, string resourceId);
+
+        Task<FhirResponse<ResourceElement>> ReadAsync(string uri);
+
+        Task<FhirResponse<ResourceElement>> VReadAsync(string resourceType, string resourceId, string versionId);
+
+        Task<FhirResponse<ResourceElement>> UpdateAsync(ResourceElement resource, string ifMatchVersion = null);
+
+        Task<FhirResponse<ResourceElement>> UpdateAsync(string uri, ResourceElement resource, string ifMatchVersion = null);
+
+        Task<FhirResponse> DeleteAsync(ResourceElement resource);
+
+        Task<FhirResponse> DeleteAsync(string uri);
+
+        Task<FhirResponse> HardDeleteAsync(ResourceElement resource);
+
+        Task<FhirResponse<ResourceElement>> SearchAsync(string resourceType, string query = null, int? count = null);
+
+        Task<FhirResponse<ResourceElement>> SearchAsync(string url);
+
+        Task<FhirResponse<ResourceElement>> SearchPostAsync(string resourceType, params (string key, string value)[] body);
+
+        Task<string> ExportAsync(Dictionary<string, string> queryParams);
+
+        Task EnsureSuccessStatusCodeAsync(HttpResponseMessage response);
+
+        Task<FhirResponse<ResourceElement>> CreateResourceElementResponseAsync(HttpResponseMessage response);
+
+        Task SetupAuthenticationAsync(TestApplication clientApplication, TestUser user = null);
+
+        Task<string> GetBearerToken(TestApplication clientApplication, TestUser user);
+
+        List<KeyValuePair<string, string>> GetAppSecuritySettings(TestApplication clientApplication);
+
+        List<KeyValuePair<string, string>> GetUserSecuritySettings(TestApplication clientApplication, TestUser user);
+
+        Task GetSecuritySettings(string fhirServerMetadataUrl);
+
+        ResourceElement GetDefaultObservation();
```

Comment:
It would be good to put these on a different interface, i.e. `IFhirTestResources`or something.